developer-workflow/c-api.rst

```diff
@@ -110,9 +110,11 @@ Guidelines for expanding/changing the public API
   fields, arguments and return values are well defined.
 
 - Functions returning ``PyObject *`` must return a valid pointer on success,
-  and ``NULL`` with an exception raised on error.
-  Most other API must return ``-1`` with an exception raised on error,
-  and ``0`` on success.
+  and ``NULL`` if and only if an exception is raised.
+  Most other API must return ``0`` on success,
```

Comment:
Which API is not in this “most” category?
```suggestion
  Other API must return ``0`` on success,
```

Reply:
I think it refers to the ones described below that return e.g. `0` for the lesser return or for "no item found", even though the wording might be a bit better.